htmd/numbautil.py

```diff
@@ -23,8 +23,62 @@ def wrapBondedDistance(d, box):
     return d
 
 
+def dihedralAngle(pos, box=None):
+    """ Calculates a dihedral angle.
+
+    Parameters
+    ----------
+    pos: np.ndarray
+        An array of 4x3 size where each row are the coordinates of an atom defining the dihedral angle
+    box: np.ndarray
+        The size of the periodic box
+
+    Returns
+    -------
+    angle: float
+        The angle in radians
+    """
+    if pos.ndim == 3 and pos.shape[2] > 1:
+        if box is None:
+            box = np.zeros((3, pos.shape[2]), dtype=pos.dtype)
+        return dihedralAngleFrames(pos, box)
+    else:
+        if box is not None:
+            box = box.squeeze()
+        return dihedralAngleFull(pos.squeeze(), box)[0]
+
+@jit(nopython=True)
+def dihedralAngleFrames(pos, box):
+    res = np.zeros(pos.shape[2], dtype=pos.dtype)
+    for f in range(pos.shape[2]):
+        res[f] = dihedralAngleFull(pos[:, :, f], box[:, f])[0]
+    return res
+
 @jit(nopython=True)
-def dihedralAngle(pos, box):
+def dihedralAngleFull(pos, box=None):
+    """ Calculates a dihedral angle.
+
+    Parameters
+    ----------
+    pos: np.ndarray
+        An array of 4x3 size where each row are the coordinates of an atom defining the dihedral angle
+    box: np.ndarray
+        The size of the periodic box
+    """
+    '''
+    http://en.wikipedia.org/wiki/Dihedral_angle#Methods_of_computation
+    https://www.cs.unc.edu/cms/publications/dissertations/hoffman_doug.pdf
+    http://www.cs.umb.edu/~nurith/cs612/hw4_2012.pdf
+
+    Ua = (A2 - A1) x (A3 - A1)
+    Ub = (B2 - B1) x (B3 - B1) = (A3 - A2) x (A4 - A2)
+    angle = arccos((Ua * Ub) / (norm(Ua) * norm(Ub)))
```

Comment:
The equation do not match the actual code!

Reply:
Yeah not anymore. It matched the old code. Removed the equation